fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -1027,8 +1027,18 @@ ACTOR Future<Void> dataDistributionRelocator(DDQueueData* self, RelocateData rd,
 					// bestTeam.second = false if the bestTeam in the teamCollection (in the DC) does not have any
 					// server that hosts the relocateData. This is possible, for example, in a fearless configuration
 					// when the remote DC is just brought up.
-					std::pair<Optional<Reference<IDataDistributionTeam>>, bool> bestTeam =
-					    wait(brokenPromiseToNever(self->teamCollections[tciIndex].getTeam.getReply(req)));
+					Future<std::pair<Optional<Reference<IDataDistributionTeam>>, bool>> fbestTeam =
+					    brokenPromiseToNever(self->teamCollections[tciIndex].getTeam.getReply(req));
+					state bool bestTeamReady = fbestTeam.isReady();
+					std::pair<Optional<Reference<IDataDistributionTeam>>, bool> bestTeam = wait(fbestTeam);
```

Comment:
Sorry for my ignorance, but would this wait not block if fbestTeam future is not ready?